records/2023-11-24.system-status.md

```diff
@@ -0,0 +1,93 @@
+| File upload | DEGRADED | The DD-API is responding to requests in more than 400ms                                |
+| File upload | DOWN     | The DD- API is not responding                                                          |
+|             |          |                                                                                        |
+|    Email    | UP       | Priority emails are sending within 60s, and Other emails are sending within 45m        |
+|    Email    | DEGRADED | Priority emails are taking longer than 60s, or Other emails are taking longer than 45m |
+|    Email    | DOWN     | No emails sent within the last 5 minutes                                               |
+|             |          |                                                                                        |
+|     SMS     | UP       | Priority SMSs are sending within 60s, and Other SMSs are sending within 45m            |
+|     SMS     | DEGRADED | Priority SMSs are taking longer than 60s, or Other SMSs are taking longer than 45m     |
+|     SMS     | DOWN     | No SMS messages sent within the last 5 minutes                                         |
+
+## Monitoring
+Each system will need to be monitored every 5m to determine which status it is in.
+
+## Systems architecture
+The website that shows system status needs to be deployed separately from the main GC Notify website so that we can continue showing status information even if the GC Notify website is down.  It should not live within our existing EKS node.
+
+### Front-end
+A new front-end website will be deployed to show the status information.  It will be served from a new domain, status.notification.canada.ca
+
+### Back-end
+A new back-end API wil be deployed that will be responsible for assessing the status of each component of the system.  
+
+#### Status checks
+- The website/API/File upload checks can all be done via simple HTTP requests to the respective systems.  No special permissions or rights are required. 
+    - **Question**: Can we leverage [the CDS status page](https://github.com/cds-snc/status-statut) for this?
+- For email and SMS, we will need to assess the actual sending time for each priority of each message type (email and SMS).  These checks will require read access to the notify database in order to calculate sending time.
+   - The healthcheck lambda should be extended to send a notification of each type and priority, and we can use these to assess the health of the email and sms sending functions.
+
+## Code architecture
+In order to maintain good developer experience, reduce code duplication and decrease maintenance, the code for the new site should be co-located in the admin repo. We may be able to use [this poetry plugin](https://pypi.org/project/poetry-multiproject-plugin/) to achieve this.  The new pages would not get deployed with admin, but rather be deployed as their own stand-alone site.
+
```

Comment:
Should we add some implementation details per-component, after discussion of course, on how we'll approach monitoring each component here, or do we want to keep it high level?

Reply:
The "Proposed component status definitions" table lays out how the statuses will be determined, but you must mean something else?

Reply:
Each component is outlined nicely in the `Status Checks` section but is fairly high level. We can add detail later, since we need to discuss this first but considering we're creating a new project for this it'd be nice to spend some time detailing a proposed project structure and per component implementation details.

Things like:
- What do we want Admin's project structure to look like? Assuming the Poetry plugin is a viable solution.
- Will the project call the API or have it's own independent DB connection?
- How we'll extend the health-check lambda

Maybe I'm missing some context or getting ahead of myself and this is the plan all along 😅.

Reply:
These are great questions!

> What do we want Admin's project structure to look like? Assuming the Poetry plugin is a viable solution.

We decided after the last meeting that we don't need to keep the same look and feel for both, we won't have to try to tie these together, which in the end will make deployment much easier.  I'll update the ADR to reflect this.

---
> Will the project call the API or have it's own independent DB connection?

It would call the DB itself as we wouldn't want to rely on API for this so that we can report the distinction when API is down but celery is still chugging along sending notifications.

---
> How we'll extend the health-check lambda

I have a one-liner on this, not sure if we need more detail or not:
> The healthcheck lambda should be extended to send a notification of each type and priority, and we can use these to assess the health of the email and sms sending functions.